### PR TITLE
Ability to specify subnet details

### PR DESCRIPTION
**Pull request by rpudlowski93**
- Added ability to specify subnet details
- Related to task #43
- Until now we could only specify number of subnets which we would like to create
- After merge of this PR we could specify list of subnets (private and public) with unique names, we could specify also address space of VPC and address prefixes of each subnet and we could specify availability zone for each subnet or choose "any" in order to chose random one which is available in the region.

### CODE REVIEW

File: docs/INPUTS.adoc [hunk 1]
```diff
@@ -11,27 +11,33 @@
...
 
-|M_PUBLIC_IPS |bool |true |no |init |If true, the EC2 instance
+|M_PUBLIC_IPS |bool |false |no |init |If true, the EC2 instance
 will have associated public IP address
 
 |M_NAT_GATEWAY_COUNT |number |1 |no |init |The number of NAT gateways
 to be created. Attached into subnets with round-robin
 
-|M_SUBNETS |map
+|M_SUBNETS |list(object)
```
Comment (atsikham) [inline docs/INPUTS.adoc:20]:
```suggestion
|M_SUBNETS |list(map)
```

File: resources/defaults.mk [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+    name: first_private_subnet,
+    availability_zone: any,
+    address_prefixes: 10.1.1.0/24
+    }],
+  public:
+  [{
+    name: first_public_subnet,
+    availability_zone: any,
+    address_prefixes: 10.1.2.0/24
+    }]
```
Comment (atsikham) [inline resources/defaults.mk:14]: check line indents

File: resources/terraform/modules/ec2/variables.tf [hunk 3]
```diff
@@ -40,24 +40,27 @@ variable "key_name" {
...
+    public = list(object({
+      name                = string
+      availability_zone   = string
+      address_prefixes    = string
+      }))
   })
   validation {
-    condition     = (var.subnets.private.count > 0 && var.subnets.public.count > 0) || var.subnets.public.count > 0
-    error_message = "At least one subnet should be created."
+    condition     = (length(var.subnets.private) > 0 && length(var.subnets.public) > 0) || length(var.subnets.public)> 0
```
Comment (atsikham) [inline resources/terraform/modules/ec2/variables.tf:62]:
```suggestion
    condition     = (length(var.subnets.private) > 0 && length(var.subnets.public) > 0) || length(var.subnets.public) > 0
```

File: resources/terraform/modules/ec2/locals.tf [hunk 4]
```diff
@@ -2,14 +2,6 @@ locals {
...
-  public_cidr_blocks        = [
-    for num in local.public_subnet_numbers:
-    cidrsubnet(var.vpc_cidr_block, 4, num)
-  ]
-  private_cidr_blocks       = [
-    for num in local.private_subnet_numbers:
-    cidrsubnet(var.vpc_cidr_block, 4, num)
-  ]
+  public_subnet_numbers     = length(var.subnets.public)
+  private_subnet_numbers    = length(var.subnets.private)
```
Comment (atsikham) [inline resources/terraform/modules/ec2/locals.tf:6]: Both locals can be removed, no need a separate variable for such values

File: resources/defaults.mk [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
-  },
-  public: {
-    count: 1
-  }
+  private:
+  [{
+    name: first_private_subnet,
+    availability_zone: any,
+    address_prefixes: 10.1.1.0/24
+    }],
```
Comment (atsikham) [inline resources/defaults.mk:8]: check line indents

File: CHANGELOG/0.0.3.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Changelog 0.0.3
+
+## [0.0.3] 2021-XX-XX
+
+### Added
+
+* [#43](https://github.com/epiphany-platform/m-aws-basic-infrastructure/issues/43) - Add the ability to specify subnet details
+
+
+### Fixed
```
Comment (atsikham) [inline CHANGELOG/0.0.3.md:10]: I see that there is no 0.0.2 release, please check that it should be so - https://github.com/epiphany-platform/m-aws-basic-infrastructure/releases

**Review by atsikham — CHANGES_REQUESTED** [review]

File: CHANGELOG/0.0.3.md (hunk 6 above)
Comment (rpudlowski93) [reply CHANGELOG/0.0.3.md:10]: Release date in changelog 0.0.2 totally mislead me... I will remove it

**Review by atsikham — APPROVED** [review]

**Review by mkyc — APPROVED** [review]
LGTM

**Pull request merged**